### PR TITLE
Solve 3.7

### PR DESCRIPTION
**Pull request by qarux**

### CODE REVIEW

File: 3_ecosystem/3_7_rand_crypto/src/main.rs [hunk 1]
```diff
@@ -1,4 +1,104 @@
+use std::path::Path;
+use argon2::{Algorithm, Argon2, Params, password_hash, PasswordHash, PasswordHasher, Version};
```
Comment (dmvict) [inline 3_ecosystem/3_7_rand_crypto/src/main.rs:2]: Please, update formatting. 
The minor change is spaces, the major change is new lines for each import. 

I can lose such cases before.

File: 3_ecosystem/3_7_rand_crypto/src/main.rs [hunk 2]
```diff
@@ -1,4 +1,104 @@
...
+use argon2::password_hash::SaltString;
+use rand::{ Rng, thread_rng };
+use sha3::{ Sha3_256, Digest };
+
+static CHAR_SET: [char; 62] = [
```
Comment (dmvict) [inline 3_ecosystem/3_7_rand_crypto/src/main.rs:7]: Please, update formatting.

File: 3_ecosystem/3_7_rand_crypto/src/main.rs [hunk 3]
```diff
@@ -1,4 +1,104 @@
...
+{
+  let mut hasher = Sha3_256::new();
+  let content = std::fs::read( path )?;
+  hasher.update( &content );
+  let mut result = [ 0u8; 32 ];
+  result.copy_from_slice( &hasher.finalize()[..] );
+  Ok( result )
+}
+
+fn hash_password( password : impl AsRef< str > ) -> password_hash::Result< String >
```
Comment (dmvict) [inline 3_ecosystem/3_7_rand_crypto/src/main.rs:49]: How do you think is the byte slice ( `AsRef< [ u8 ] >` ) better input type?

File: 3_ecosystem/3_7_rand_crypto/src/main.rs [hunk 4]
```diff
@@ -1,4 +1,104 @@
...
+{
+  let salt = SaltString::generate( &mut thread_rng() );
+  // Values from OWASP cheatsheet
+  let password_hash = Argon2::new(
```
Comment (dmvict) [inline 3_ecosystem/3_7_rand_crypto/src/main.rs:53]: Please, update formatting.

File: 3_ecosystem/3_7_rand_crypto/src/main.rs [hunk 5]
```diff
@@ -1,4 +1,104 @@
...
+    Algorithm::Argon2id,
+    Version::V0x13,
+    Params::new( 15000, 2, 1, Some( 32 ) ).unwrap(),
+  )
+    .hash_password( password.as_ref().as_bytes(), &salt )?;
```
Comment (dmvict) [inline 3_ecosystem/3_7_rand_crypto/src/main.rs:58]: Please, update formatting.

**Comment by qarux** [comment]
All fixed

File: 3_ecosystem/3_7_rand_crypto/src/main.rs [hunk 6]
```diff
@@ -1,4 +1,113 @@
+use std::path::Path;
+use argon2::Algorithm;
+use argon2::Argon2;
+use argon2::Params;
```
Comment (dmvict) [inline 3_ecosystem/3_7_rand_crypto/src/main.rs:4]: In the future, it is better to use this syntax : 
```rs
use dep::
{
  import1,
  import2, 
  import3,
};

**Comment by dmvict** [comment]
Can be merged.

**Pull request merged**